File: nodes/string-substring/string-substring.process.py
async def process(inputs, settings, config, nodeConfig):
  text = str(inputs.get("text", ""))
  start = int(inputs.get("start", 0))
  end = int(inputs.get("end")) if inputs.get("end") is not None else None
  
  return {
    "text": text[start:end]
  } 

File: sdks/nodejs/nodes/string-substring/string-substring.process.js
export default async ({inputs, settings, config}) => {

  const text = String(inputs.text || "");
  const start = parseInt(inputs.start) || 0;
  const end = inputs.end !== undefined ? parseInt(inputs.end) : undefined;
  
  return { 
    text: text.slice(start, end)
  };
}; 